rskj-core/src/test/java/co/rsk/peg/feeperkb/FeePerKbIntegrationTest.java

```diff
@@ -0,0 +1,567 @@
+package co.rsk.peg.feeperkb;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.core.RskAddress;
+import co.rsk.peg.InMemoryStorage;
+import co.rsk.peg.feeperkb.constants.FeePerKbConstants;
+import co.rsk.peg.feeperkb.constants.FeePerKbMainNetConstants;
+import co.rsk.peg.storage.StorageAccessor;
+import org.ethereum.core.SignatureCache;
+import org.ethereum.core.Transaction;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@TestInstance(Lifecycle.PER_CLASS)
```

Comment:
Not sure, but I don't think the `@TestInstance(Lifecycle.PER_CLASS)` annotation is not needed here since it is implied by `@TestMethodOrder(MethodOrderer.OrderAnnotation.class)`

Reply:
Hey, @apancorb!
@TestMethodOrder(MethodOrderer.OrderAnnotation.class) doesn't have built-in @TestInstance(Lifecycle.PER_CLASS) in it. If we don't set `@TestInstance(Lifecycle.PER_CLASS)` it goes work `Lifecycle.PER_METHOD` by default. That means, if we are going to share a state of a variable in all methods, for instance, `currentFeePerKb`, it must be static. Apart from that, the method annotated with `@BeforeAll` must be static and all into it, static as well. That being said, I believe our best option is to use `@TestInstance(Lifecycle.PER_CLASS)`.

Reply:
Thanks!